### PR TITLE
Create status report text file

### PR DESCRIPTION
**Pull request by adamnitti**
## overall purpose
added status report text file to output

### CODE REVIEW

File: csv-bulk-signature/src/main.ts [hunk 1]
```diff
@@ -39,7 +39,7 @@ interface TemplateData {
 }
 
 const skippedRows: string[] = [];
-let counter = 0;
+const processedRows: string[] = [];
```
Comment (adamnitti) [inline csv-bulk-signature/src/main.ts:42]: added explicit array for signatures processed successfully to access data for status report

File: csv-bulk-signature/src/main.ts [hunk 2]
```diff
@@ -75,7 +75,8 @@ function generateSignatures(
       const hasRequiredData = checkRequiredFields(row);
       if (!hasRequiredData) {
         skippedRows.push(fullName);
-        return;
+      } else {
+        processedRows.push(fullName);
```
Comment (adamnitti) [inline csv-bulk-signature/src/main.ts:79]: pushing rows processed to array

File: csv-bulk-signature/src/main.ts [hunk 3]
```diff
@@ -94,20 +95,13 @@ function generateSignatures(
...
-
       // show results summary
-      console.log("Number of signatures processed", counter);
-      console.log(
-        "Number of signatures skipped",
-        skippedRows.length,
-        "for missing required fields (scroll up to see list)"
-      );
+      console.log("Number of signatures processed", processedRows.length);
+      console.log("Number of signatures skipped", skippedRows.length);
```
Comment (adamnitti) [inline csv-bulk-signature/src/main.ts:103]: count summaries for signatures

File: csv-bulk-signature/src/main.ts [hunk 4]
```diff
@@ -94,20 +95,13 @@ function generateSignatures(
...
+      createStatusReport(); // refactoring the results specifics to be shown in a text file, instead
```
Comment (adamnitti) [inline csv-bulk-signature/src/main.ts:104]: calling function for creating status report text file

File: csv-bulk-signature/src/main.ts [hunk 5]
```diff
@@ -133,7 +127,7 @@ function deleteAllSignatures() {
     }
 
     for (const file of files) {
-      if (path.extname(file) === ".html") {
+      if (path.extname(file) === ".html" || path.extname(file) === ".htm") {
```
Comment (adamnitti) [inline csv-bulk-signature/src/main.ts:130]: adding `.htm` for signature deletion

File: csv-bulk-signature/src/main.ts [hunk 6]
```diff
@@ -194,5 +188,18 @@ async function zipUpFile() {
   });
 
   const buffer = await zip.generateAsync({ type: "nodebuffer" });
-  fs.writeFileSync(`signatures-${new Date().toDateString()}.zip`, buffer);
+  fs.writeFileSync(
+    `signatures-${new Date().toISOString().split("T")[0]}.zip`,
```
Comment (adamnitti) [inline csv-bulk-signature/src/main.ts:192]: updated date stamp

File: csv-bulk-signature/src/main.ts [hunk 7]
```diff
@@ -194,5 +188,18 @@ async function zipUpFile() {
...
+    buffer
+  );
+}
+
+function createStatusReport() {
```
Comment (adamnitti) [inline csv-bulk-signature/src/main.ts:197]: function for creating new status report text file

**Review by chancesmith — APPROVED** [review]

**Pull request merged**